app/src/modules/whyis-dataset.js

```diff
@@ -0,0 +1,282 @@
+  depiction: `${foaf}depiction`,
+  hasContent: 'http://vocab.rpi.edu/whyis/hasContent',
+  accessURL: `${dcat}accessURL`
+}
+
+const datasetPrefix = 'dataset'
+
+// Generate a randum uuid, or use current if exists
+function generateDatasetId (guuid) {
+  var datasetId
+  if (arguments.length === 0) {
+    const { v4: uuidv4 } = require('uuid')
+    datasetId = uuidv4()
+  } else {
+    datasetId = guuid
+  }
+  return `${lodPrefix}/explorer/${datasetPrefix}/${datasetId}`
+}
+
+function buildDatasetLd (dataset) {
+  dataset = Object.assign({}, dataset)
+  dataset.context = JSON.stringify(dataset.context)
+  const datasetLd = {
+    '@id': dataset.uri,
+    '@type': []
+  }
+
+  if (dataset['@type'] != null) {
+    datasetLd['@type'].push(dataset['@type'])
+  }
+
+  Object.entries(dataset)
+    // filter out the ones that aren't in our allowed fields
+    .filter(([field, value]) => datasetFieldUris[field.toLowerCase()])
+    .forEach(([field, value]) => {
+      // make a new dictionary
+      var ldValues = {}
+      // If the field has a value
+      if (!isEmpty(value)) {
+        ldValues = recursiveFieldSetter([field, value])
+        datasetLd[datasetFieldUris[field.toLowerCase()]] = ldValues
+      }
+    })
+  return datasetLd
+}
+
+// Recursively check if a value is empty
+function isEmpty (value) {
+  // Base case
+  if ((value === '') || (value === null) || (value === []) || (value === 'undefined')) {
+    return true
+  } else if (Array.isArray(value)) {
+    // Is empty if array has length 0
+    let arrayEmpty = (value.length === 0)
+    for (var val in value) {
+      // if any entry in the array is empty, it's empty
+      arrayEmpty = arrayEmpty || isEmpty(value[val])
+    }
+    return arrayEmpty
+  } else if (typeof (value) === 'object') {
+    let objEmpty = false
+    for (var property in value) {
+      // if any attribute of the object is empty, it's empty
+      objEmpty = objEmpty || isEmpty(value[property])
+    }
+    return objEmpty
+  }
+  return false
+}
+
+// Helper for assigning values into JSON-LD format
+function recursiveFieldSetter ([field, value]) {
+  // If the value is also an array, recur through the value
+  if (Array.isArray(value)) {
+    var fieldArray = []
+    for (const val in value) {
+      fieldArray.push(recursiveFieldSetter([field, value[val]]))
+    }
+    return fieldArray
+  } else {
+    var fieldDict = {}
+    // Fields may have multiple values, so loop through all
+    for (const val in value) {
+      // type, value and id aren't in datasetFieldURIs dictionary
+      // but they are valid keys, so set the value to their value
+      if ((val === '@type') || (val === '@value') || (val === '@id')) {
+        fieldDict[val] = value[val]
+        // but if the value of val is an allowed field, use the field's value
+        // e.g., type = organization, and organization -> foaf:Organization
+        if (Object.prototype.hasOwnProperty.call(datasetFieldUris, value[val].toLowerCase())) {
```

Comment:
Not entirely sure, but this had been deprecated or will be soon. Can we try `Object.getOwnPropertyDescriptor`

Reply:
(Noting for my own reference that I've addressed this one locally)